bindings/pyroot/pythonizations/test/rbatchgenerator_completeness.py

```diff
@@ -0,0 +1,229 @@
+import unittest
+import os
+import ROOT
+
+class RBatchGeneratorMultipleFiles(unittest.TestCase):
+
+    file_name1 = "first_half.root"
+    file_name2 = "second_half.root"
+    tree_name = "mytree"
+
+    # Helpers
+    def create_10_entries_file(self):
+        df2 = ROOT.RDataFrame(10)\
+            .Define("b1", "(int) rdfentry_")\
+            .Define("b2", "(double) b1*b1")\
+            .Snapshot(self.tree_name, self.file_name1)
+    
+    # def create_20_entries_file(self):
+    #     df1 = ROOT.RDataFrame(20)\
+    #         .Define("b1", "(int) rdfentry_ + 10")\
+    #         .Define("b2", "(double) rdfentry_ * rdfentry_")\
+    #         .Snapshot(self.tree_name, self.file_name2)
+        
+        #print(df1.Describe())
+        #print(df2.Describe())
+
+        #df2.Display("",).Print()
+
+
+    def teardown_file(self, file):
+        os.remove(file)
+
+
+    def test01_each_element_is_generated_unshuffled(self):
+        self.create_10_entries_file()
+
+        gen_train, gen_validation = ROOT.TMVA.Experimental.CreateNumPyGenerators(
+        tree_name=self.tree_name,
+        file_name=self.file_name1,
+        batch_size=3,
+        chunk_size=5,
+        target="b2",
+        validation_split=0.3,
+        shuffle=False,
+        drop_remainder=False
+        )
+
+        results_x_train = [2.0, 3.0, 4.0, 7.0, 8.0, 9.0]
+        results_x_val = [0.0, 1.0, 5.0, 6.0]
+        results_y_train = [4.0, 9.0, 16.0, 49.0, 64.0, 81.0]
+        results_y_val = [0.0, 1.0, 25.0, 36.0]
+        
+        collected_x_train = []
+        collected_x_val = []
+        collected_y_train = []
+        collected_y_val = []
+
+        for x, y in gen_train:
+            collected_x_train.append(x.tolist())
+            collected_y_train.append(y.tolist())
+        
+        for x, y in gen_validation:
+            collected_x_val.append(x.tolist())
+            collected_y_val.append(y.tolist())
+
+        flat_x_train = [x for xl in collected_x_train for xs in xl for x in xs]
+        flat_x_val = [x for xl in collected_x_val for xs in xl for x in xs]
+        flat_y_train = [y for yl in collected_y_train for ys in yl for y in ys]
+        flat_y_val = [y for yl in collected_y_val for ys in yl for y in ys]
```

Comment:
As we discussed offline, these tests must also check the shape of every batch to be correct

Reply:
Already done for the next PR